pkg/apis/projectcalico/v3/nodestatus.go

```diff
@@ -0,0 +1,178 @@
+	AdditionalInfo string `json:"additionalInfo,omitempty"`
+
+	// BGPStatus holds node BGP status.
+	BGPStatus CalicoNodeBGPStatus `json:"bgpStatus,omitempty"`
+}
+
+// CalicoNodeBGPStatus defines the observed state of BGP status on the node.
+type CalicoNodeBGPStatus struct {
+	// The total number of established bgp sessions.
+	NumEstablished int `json:"numEstablished,omitempty"`
+
+	// The total number of non-established bgp sessions.
+	NumNotEstablished int `json:"numNotEstablished,omitempty"`
+
+	// BirdStatus represents the latest observed status of bird.
+	BirdStatus CalicoNodeBirdStatus `json:"birdStatus,omitempty"`
+
+	// Peers represents BGP peers status on the node.
+	Peers []CalicoNodePeers `json:"peers,omitempty"`
+
+	// Routes represents routes on the node.
+	Routes []CalicoNodeRoutes `json:"routes,omitempty"`
+}
+
+// CalicoNodeBirdStatus defines the observed state of bird.
+type CalicoNodeBirdStatus struct {
+	// Ready indicates if bird status is ready.
+	Ready bool `json:"ready,omitempty"`
+
+	// Bird version.
+	Version string `json:"version,omitempty"`
+
+	// Route ID used by bird.
+	RouteID string `json:"routeID,omitempty"`
+
+	// ServerTime holds the value of serverTime from birdctl output.
+	ServerTime string `json:"serverTime,omitempty"`
+
+	// LastBootTime holds the value of lastBootTime from birdctl output.
+	LastBootTime string `json:"lastBootTime,omitempty"`
+
+	// LastReconfigTime holds the value of lastReconfigTime from birdctl output.
+	LastReconfigTime string `json:"lastReconfigTime,omitempty"`
+}
+
+// CalicoNodePeers contains the status of BGP peers on the node.
+type CalicoNodePeers struct {
+	// IP address of the peer whose condition we are reporting.
+	// If port number is given, format should be `[<IPv6>]:port` or `<IPv4>:<port>` for IPv4.
+	// If optional port number is not set, and this peer IP and ASNumber belongs to a calico/node
+	// with ListenPort set in BGPConfiguration, then we use that port to peer.
+	PeerIP string `json:"peerIP,omitempty"`
+
+	// The type is type of bgp session state.
+	Type BGPPeerType `json:"type,omitempty"`
+
+	// The state is the bgp session state.
+	State string `json:"state,omitempty"`
+
+	// Since is the time since the condition last changed.
+	Since string `json:"since,omitempty"`
+
+	// The reason it's in the current state.
+	Reason string `json:"info,omitempty"`
+}
+
+// CalicoNodePeers contains the status of BGP routes on the node.
+type CalicoNodeRoutes struct {
+	// Destination of the route.
+	Destination string `json:"destination,omitempty"`
+
+	// Gateway for the destination.
+	Gateway string `json:"gateway,omitempty"`
+
+	// Interface for the destination
+	Interface string `json:"interface,omitempty"`
+
+	// InstalledBy indicates who installed this route.
+	// If it is populated by a BGP peer, this is the name of the BGPPeer object.
+	// If it is populated by node mesh, this is the name of the node.
+	InstalledBy string `json:"installedBy,omitempty"`
```

Comment:
"LearnedFrom" rather than "installedBy"?

installed seems to suggest programming of the route locally, whereas this field actually represents where the local node learned the route from.

Also, as suggested, not all learned routes may be installed.